pkg/controllers/externalsecret/metrics.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalsecret
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	ExternalSecretSubsystem = "external_secret"
+	SyncCallsKey            = "sync_calls_total"
+	SyncCallsErrorKey       = "sync_calls_error"
+)
+
+var (
+	syncCallsTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
A few comments on these metrics:

1. `sync_calls_total`, we probably want to drop the name/namespace cardinality and just have this metric track the count of sync calls for the controller irrespective of which resource the call was for.
2. `sync_calls_error`, a metric similar to these seems useful for increasing observability for this operator, but we should follow how [Kube-state-metric](https://github.com/kubernetes/kube-state-metrics/blob/v1.9.8/docs/deployment-metrics.md) and [cert-manager](https://github.com/jetstack/cert-manager/blob/v1.3.0/pkg/metrics/metrics.go) do this and change this a metric which tracks the status condition for each external secret. A gauge of `externalsecret_status_condition` which has cardinality of name/namespace/condition might allow better tracking of the state of the system.